.github/scripts/ci/detect-changes.sh

```
@@ -0,0 +1,102 @@
+  echo "[detect-changes] No changed files between BASE_SHA and HEAD — leaving all flags false."
+  {
+    echo "src=false"
+    echo "md=false"
+    echo "docs=false"
+    echo "pipeline_ts=false"
+  } >> "$GITHUB_OUTPUT"
+  exit 0
+fi
+
+echo "[detect-changes] ${BASE_SHA:0:12}...HEAD changed files:"
+printf '  - %s\n' ${changed_files}
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)